catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/QueryFilterTransformerProvider.java

```diff
@@ -0,0 +1,114 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.spatial.ogc.csw.catalog.endpoint;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.xml.namespace.QName;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.transform.QueryFilterTransformer;
+
+/**
+ * Manages a reference list of {@link QueryFilterTransformer}'s by mapping
+ * them to the {@link QName}'s they apply to.
+ */
+public class QueryFilterTransformerProvider {
+    private static final Logger LOG = LoggerFactory.getLogger(QueryFilterTransformerProvider.class);
+
+    private Map<QName, QueryFilterTransformer> queryFilterTransformerMap =
+            new ConcurrentHashMap<>();
+
+    public synchronized void bind(ServiceReference<QueryFilterTransformer> reference) {
+        if (reference == null) {
+            return;
+        }
+
+        List<QName> namespaces = getNamespaces(reference);
+        QueryFilterTransformer transformer = getTransformer(reference);
+
+        for (QName namespace : namespaces) {
+            queryFilterTransformerMap.put(namespace, transformer);
+        }
+    }
+
+    public synchronized void unbind(ServiceReference<QueryFilterTransformer> reference) {
+        if (reference == null) {
+            return;
+        }
+
+        try {
+            List<QName> namespaces = getNamespaces(reference);
+
+            for (QName namespace : namespaces) {
+                queryFilterTransformerMap.remove(namespace);
+            }
+        } finally {
+            getBundleContext().ungetService(reference);
+        }
+    }
+
+    public QueryFilterTransformer getTransformer(QName qName) {
+        return queryFilterTransformerMap.get(qName);
+    }
+
+    private List<QName> getNamespaces(ServiceReference<QueryFilterTransformer> reference) {
+        Object id = reference.getProperty("id");
+        List<QName> result = new ArrayList<>();
+        if (id instanceof List) {
+            List<String> namespaces = (List<String>) id;
+            for (String namespace : namespaces) {
+                result.add(QName.valueOf(namespace));
+            }
+        } else if (id instanceof String) {
+            result.add(QName.valueOf((String) id));
+        } else {
+            LOG.error(
```

Comment:
❓ Do we expect the admin to be able to do something about this? If not, error is probably too high (see https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels).